### PR TITLE
Common input pipeline for single- and multi-band imagery and AOI object for input data

### LINKED ISSUE #223: Reading single-band rasters: unified approach with multiband reader
As operationalization approaches, inference.py will need to read and concatenate single band rasters rather than [multiband only](https://github.com/NRCan/geo-deep-learning/blob/develop/inference.py#L120). This implies extending [sample_creation.py](https://github.com/NRCan/geo-deep-learning/blob/develop/sample_creation.py)'s use to inference.py and possibly merging with [images_to_samples.py](https://github.com/NRCan/geo-deep-learning/blob/develop/images_to_samples.py) (soon to become data_to_tiles.py --> #215 ). 

This issue is closely related to #221 and #222. A future reading utility will have to manage a list of rasters as input with variable configration (some may be multiband, others will be single-band, maybe even for a same training dataset). This utility will then be available for tiling as well as inference.

### PR DESCRIPTION
**Pull request by remtav**
aoi.py: 
- create an AOI object with input validation. AOI would be the core input for tiling, training and inference, though only yet implemented for tiling.
- add stac item support
geoutils.py:
- add utils: is_stac_item, stack_vrts() for create artificial multi-band raster from single-bands files
test_aoi.py: add first test for parsing raster input from 3 types to a single rasterio.RasterDataset object
default.yaml: activate debug functionality for logging
test_ci_segmentation_multiclass.yaml: replace 'modalities' with 'bands' key
test_ci_segmentation_binary.yaml: idem
sample_creation.py: delete
utils.py: remove validation from read_csv() function.

fixes #223

### CODE REVIEW

**Review by CharlesAuthier — APPROVED** [review]

File: dataset/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+| my_dir/my_geoimagery2.tif | my_dir/my_geogt2.gpkg    | tst           | NewBrunswick-23   |
+| ...                       | ...                      | ...           | ...               |
+
+> The use of aoi id information will be implemented in a near future. It will serve, for example, to print a detailed report of sampling, training and evaluation, or for easier debugging.
+
+The path to a custom csv must be entered in the [dataset configuration](https://github.com/NRCan/geo-deep-learning/blob/develop/config/dataset/test_ci_segmentation_binary.yaml#L9). See the [configuration documentation](config/README.md) for more information.
+Also check the [suggested folder structure](https://github.com/NRCan/geo-deep-learning#folder-structure).
+
+## Dataset splits
+Split in csv should be either "trn", "tst" or "inference". The validation split is automatically created during sampling. It's proportion is set by the [dataset config](https://github.com/NRCan/geo-deep-learning/blob/develop/config/dataset/test_ci_segmentation_binary.yaml#L8).
```
Comment (mpelchat04) [inline dataset/README.md:17]: ... automatically created from the "trn" split, during sampling.

File: dataset/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Raster and vector file compatibility
+Rasters to be used must be in a format compatible with [rasterio](https://rasterio.readthedocs.io/en/latest/quickstart.html?highlight=supported%20raster%20format#opening-a-dataset-in-reading-mode)/[GDAL](https://gdal.org/drivers/raster/index.html) (ex.: GeoTiff). Similarly, labels (aka annotations) for each image must be stored as polygons in a [Geopandas compatible vector file](Rasters to be used must be in a format compatible with [rasterio](https://rasterio.readthedocs.io/en/latest/quickstart.html?highlight=supported%20raster%20format#opening-a-dataset-in-reading-mode)/[GDAL](https://gdal.org/drivers/raster/index.html) (ex.: GeoTiff). Similarly, labels (aka annotations) for each image must be stored as polygons in a [Geopandas compatible vector file](https://geopandas.org/en/stable/docs/user_guide/io.html#reading-spatial-data) (ex.: GeoPackage).
+) (ex.: GeoPackage).
+
+## Single-band vs multi-band imagery
```
Comment (mpelchat04) [inline dataset/README.md:23]: Neat!

File: dataset/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+During execution, this would result in using, **in the same order as bands appear in dataset config**, the following files:
+`my_dir/my_singleband_geofile_band_R.tif`
+`my_dir/my_singleband_geofile_band_G.tif`
+`my_dir/my_singleband_geofile_band_B.tif`
+
+> To simplify the use of both single-band and multi-band rasters through a unique input pipeline, single-band files are artificially merged as a [virtual raster](https://gdal.org/drivers/raster/vrt.html).
+
+### 3. Path to a Stac Item 
+> Only Stac Items referencing **single-band assets** are supported currently. See [our Worldview-2 example](https://datacube-stage.services.geo.ca/api/collections/spacenet-samples/items/SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03).
```
Comment (mpelchat04) [inline dataset/README.md:54]: WorldView-3

**Review by mpelchat04 — APPROVED** [review]

File: dataset/README.md (hunk 1 above)
Comment (remtav) [reply dataset/README.md:17]: Thanks, I'll correct this in a next PR.

**Review by CharlesAuthier — APPROVED** [review]

**Review by valhassan — APPROVED** [review]

**Pull request merged**